src/content/en/2021/caching.md

```diff
@@ -17,4 +20,425 @@ featured_stat_3: TODO
+
+Between 2020 and 2021, the percent using invalid date improved by a .5% but got worse for last-modified and expires showing that it was related to how the date was set on caching.
+
+### Vary
+
+An essential step in caching a resource is understanding if it was already previously cached. The browser usually uses the url as the cache key. At the same time, requests with similar urls but different `accept-encoding` could be cached incorrectly. That's why we use the Vary header to instruct the browser to add a value of one or more headers to the cache key.
+
+The most popular Vary header is `accept-encoding` with 90.3% usage, followed by `user-agent` with 10.9%, `origin` with 10.1%, and `accept` with 4.8%. 
+
+[TODO Add figure showing Vary header adoption across all requests (pct_using_vary) and cache-control (pct_of_vary_with_cache_control), data from https://docs.google.com/spreadsheets/d/1-v3yR0LZIC3t4zWtqTgR3jJsKjjRMP-HATU2caP8e2c/edit#gid=1033782866 and https://docs.google.com/spreadsheets/d/1fYmpSN3diOiFrscS75NsjfsrKXzxxhUMNcYSqXnQJQU/edit#gid=1115686547 Potentiall we mighe want to show 2020 vs 2021]
+
+It's important to point out that only 46.25% of total requests audited use the Vary header. Of these requests, only 83.4% also have the cache-control. 
+
+Compared to 2020, we see an overall increase of 2.85% in use of the Vary header.
+
+## Setting cookies on cacheable responses
+
+In the 2020 Caching chapter, we were reminded to be aware of using `set-cookie` with cacheable responses because only 4.9% of responses used the `private` directive, putting a user's private data at risk of accidentally being accidentally served to a different user via a CDN.
+
+[TODO Add figure showing Set-Cooike usage on cacheable responses, similar to https://20211128t100318-dot-webalmanac.uk.r.appspot.com/static/images/2020/caching/set-cookie-usage-on-cacheable-responses.png, and data from https://docs.google.com/spreadsheets/d/1-v3yR0LZIC3t4zWtqTgR3jJsKjjRMP-HATU2caP8e2c/edit#gid=104392613 Potentiall we mighe want to show 2020 vs 2021]
+
+In 2021, we see an increase in awareness regarding set-cookie and caching coexisting. While still only 5% of web pages are using the private directive with Set-Cookie, the total number of cacheable set-cookie responses decreased by 4.41%.
+
+## What type of content are we caching?
+
+We know that Font, CSS, and Audio files are over 99% cacheable, and almost 100% of pages are caching fonts. With CSS, audio, and video being some of the most common static files on web pages, they are also highly cacheable.
```

Comment:
Edits: 

We see that Font, CSS, and Audio files are over 99% cacheable, with almost 100% of pages currently caching fonts. This could be in part due to their static nature, making them prime choices for caching because 

{{ figure_markup(
  image="caching-by-resource-type.png",
  caption="The percent of requests that use caching strategies by resource type.",
  description="TODO",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=2036781114&format=interactive",
  height=436,
  sheets_gid="1202769738",
  sql_file="non_cacheable_by_resource_type.sql"
) }}

However, some of our go-to resources are non-cacheable, likely due to their dynamic nature. Notably, HTML saw the highest percentage of non-cacheable resources at 23.4%, followed closely by Images with 10.1%. 

When we compare the mobile data from 2020 to 2021, we notice a 5.1% increase in cacheable HTML. This tells us we may be moving towards better usage of our CDNs to cache HTML pages, like those rendered by an SSRt that tdeto change less frequently. Pages are typically generated by Server-Side Rendered (SSR) applications if the content of a particular web page doesn't change frequently. The url can potentially serve the same HTML for weeks or even months, making that content highly cacheable.

Taking a look at the Median TTL across all resource types, we see that even if we are caching similar percentage in total, there is a much lengthier cache for mobile, particularly for HTML, audio and video.